### PR TITLE
Added support for title in each option 

### PR DESCRIPTION
**Pull request by surbhimahajan-wingify**
Feature: Added Support for title in each option in both array and object data sources cases

### CODE REVIEW

File: chosen.js [hunk 1]
```diff
@@ -90,6 +90,27 @@
...
+                options are rendered again and removed all titles from options
+                TODO: will debug it to find out reason */
+                $timeout(function() {
+                    // Convert object into array first and then assign titles to all options according to values of object
+                    if(!(options instanceof Array)) {
+                        options = Object.keys(options).map(function(key) {
+                            return options[key]
+                        })
+                    }
+                    for (var index=options.length - 1,lastOptionIndex =  $(element).find("option").length - 1;index>= 0;index--,lastOptionIndex--){
```
Comment (akanshgulati) [inline chosen.js]: Can we store the element `$(element).find("option").length` in a variable as its used below too.

File: chosen.js [hunk 2]
```diff
@@ -90,6 +90,27 @@
           if (attr.ngOptions && ngModel) {
             match = attr.ngOptions.match(NG_OPTIONS_REGEXP);
             valuesExpr = match[7];
+            var options = scope.$eval(valuesExpr);
+            // Adding title in case of array of options and object data source
+            if (options && typeof options === 'object') {
```
Comment (akanshgulati) [inline chosen.js]: typeof is not that appropriate when dealing with objects or arrays.

Comment (surbhimahajan-wingify) [reply chosen.js]: As discussed with you, typeof can be used here

File: chosen.js (hunk 1 above)
Comment (surbhimahajan-wingify) [reply chosen.js]: done

**Review by softvar — APPROVED** [review]

**Review by dinkar — APPROVED** [review]

**Pull request merged**